src/config.rs

```diff
@@ -87,78 +148,71 @@ pub struct Config {
-                      .expect("No options.girs_dir in config")
-                      .as_str().unwrap(),
+        let girs_dir: PathBuf = match args.get_str("-d") {
+            "" => try!(toml.lookup_str("options.girs_dir", "No options.girs_dir in config", &config_file)),
             a => a
-        };
+        }.into();
 
         let (library_name, library_version) =
             match (args.get_str("<library>"), args.get_str("<version>")) {
             ("", "") => (
-                toml.lookup("options.library")
-                    .expect("No options.library in config")
-                    .as_str().unwrap(),
-                toml.lookup("options.version")
-                    .expect("No options.version in config")
-                    .as_str().unwrap()
+                try!(toml.lookup_str("options.library", "No options.library in config", &config_file)),
+                try!(toml.lookup_str("options.version", "No options.version in config", &config_file))
             ),
-            ("", _) | (_, "") => panic!("Library and version can not be specified separately"),
+            ("", _) | (_, "") => try!(Err(("Library and version can not be specified separately", &config_file))),
             (a, b) => (a, b)
         };
 
-        let target_path = match args.get_str("-o") {
-            "" => toml.lookup("options.target_path")
-                    .expect("No target path specified")
-                    .as_str().unwrap(),
+        let target_path: PathBuf = match args.get_str("-o") {
+            "" => try!(toml.lookup_str("options.target_path", "No target path specified", &config_file)),
             a => a
-        };
+        }.into();
 
         let mut objects = toml.lookup("object").map(|t| gobjects::parse_toml(t))
             .unwrap_or_else(|| Default::default());
         gobjects::parse_status_shorthands(&mut objects, &toml);
 
-        let external_libraries = toml.lookup("options.external_libraries")
-            .map(|a| a.as_slice().unwrap().iter()
-                .filter_map(|v|
+        let external_libraries = match toml.lookup("options.external_libraries") {
+            Some(a) => try!(a.as_result_slice("options.external_libraries", &config_file))
+                .iter().filter_map(|v|
                     if let &toml::Value::String(ref s) = v { Some(s.clone()) } else { None } )
-                .collect())
-            .unwrap_or_else(|| Vec::new());
+                .collect(),
+            None => Vec::new(),
+        };
 
-        let min_cfg_version = toml.lookup("options.min_cfg_version")
-            .map_or_else(|| Ok(Default::default()), |t| t.as_str().unwrap().parse())
-            .unwrap_or_else(|e| panic!(e));
+        let min_cfg_version = match toml.lookup("options.min_cfg_version") {
+            Some(v) => try!(try!(v.as_result_str("options.min_cfg_version", &config_file))
+                            .parse().map_err(|e: String| (e, config_file))),
```

Comment:
This should probably be unrolled into a block with two `try!` lines, it's too difficult to parse.